### PR TITLE
Fixes for DDR

### PR DESCRIPTION
**Pull request by youngar**
Continuing from #1014.

This PR introduces the following changes:
- Code reorganization
- Cmake compilation support
- Two phase superset generation:
  - generate IL for all compilation units
  - combine generated IL into one cohesive IL
- Many small improvements to better support IL generation for different platforms and compilers 
- Building DDR in our CI

### CODE REVIEW

**Comment by mstoodle** [comment]
Uh, any update on what's going on in this PR? :) @youngar ?

File: ddr/include/ddr/ir/Modifiers.hpp [hunk 1]
```diff
@@ -19,7 +19,7 @@
 #ifndef MODIFIERS_HPP
 #define MODIFIERS_HPP
 
-#include <string>
+#include "ddr/std/string.hpp"
 #include <vector>
```
Comment (rwy0717) [inline ddr/include/ddr/ir/Modifiers.hpp]: Does vector also need a compatibility header?

File: ddr/include/ddr/ir/Type.hpp [hunk 2]
```diff
@@ -22,7 +22,7 @@
 #include "ddr/config.hpp"
 
 #include <set>
```
Comment (rwy0717) [inline ddr/include/ddr/ir/Type.hpp]: And set too?

File: ddr/include/ddr/ir/Modifiers.hpp (hunk 1 above)
Comment (jason-hall) [inline ddr/include/ddr/ir/Modifiers.hpp]: I can't think of a reason it would

**Comment by youngar** [comment]
I have created a CQ and uploaded a patch for this change.  For those who are members of the project, the CQ is here: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=13962.  I will try to keep this PR updated with the CQ status.

**Comment by youngar** [comment]
This CQ has been approved!

**Pull request merged**